src/main/io/gps.c

```diff
@@ -688,79 +1079,156 @@ void gpsInitUblox(void)
 
             if (gpsData.ackState == UBLOX_ACK_IDLE) {
                 switch (gpsData.state_position) {
+                    case UBLOX_DETECT_UNIT:
+                        if (!gpsData.acquiredMonVer) {
+                            ubloxSendClassMessage(CLASS_MON, MSG_MON_VER, 0);
+                            break;
+                        }
+                        gpsData.state_position++;
+                        break;
                     case UBLOX_INITIALIZE:
-                        gpsData.ubloxUsePVT = true;
+                        gpsData.updateRate = gpsConfig()->gps_update_rate_hz;
+
+                        ubloxSetNavRate(1, 1, 1);
+
+                        gpsData.ubloxUsePVT = gpsData.unitVersion > UBX_VERSION_M6;
                         gpsData.ubloxUseSAT = true;
+
                         ubloxSendNAV5Message(gpsConfig()->gps_ublox_acquire_model);
+
+                        break;
+                    case UBLOX_CFG_ANA:
+                        if (gpsData.unitVersion > UBX_VERSION_M7) {
+                            ubloxSendNav5XMessage();
+                            break;
+                        }
+                        gpsData.state_position++;
+                        break;
+                    case UBLOX_MSG_DISABLE_NMEA:
+                        if (ubloxHasValSetGet()) {
```

Comment:
I believe this conditional is reversed, and will result in a bug.

ubloxHasValSetGet( ) returns true if the protocol version is greater than 23.01. These more recent protocols **_DO_** support the new UBX-CFG-VALSET command for configuring items (and deprecate/remove the old commands).

The way this is written, if it returns true and we're using the new protocols, it will move to the next state, which starts disabling the NMEA messages using the **_OLD_** config commands (that would no longer be supported in the new protocol), and breaks out of the rest of this case so it never disables them with the new config command.

If it returns false and we're using the old protocols, it will disable the NMEA messages using the **_NEW_** UBX-CFG-VALSET command, and then skip over all the states to disable using the old commands.

So, I think this conditional needs to be reversed to:
```if (!ubloxHasValSetGet()) {```

Reply:
I think this has been resolved, please confirm @ZzyzxTek, thanks.

Reply:
@ctzsnooze Yes, this was changed to use the `ubloxM9orAbove` check, if so it does the VALSET and skips the CFG-MSG states, otherwise it skips the VALSET and does them. Resolved!